### PR TITLE
Add Reset button to login screen 

### PR DESCRIPTION
**Pull request by e-wai**
## Notion ticket link
[Allow users to reset their password](https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=4a47edc475b34bb99b25ff1fae2a5507&p=92887fb5e64845e882c4c79a97a18ff6)

## Implementation description
* Add reset button to home screen
* Implement basic logic for parsing email string
* Create alerts based on whether the reset email was successfully sent
* Remove authentication requirement that was needed to send reset email

## Steps to test
1. Try to reset password from login page with legitimate email form (eg. `test@gmail.com`), random string (eg. `potato`), and real email (eg. `planetread+user@uwblueprint.org`). Should receive email for the latter
2. Login fully and try to reset password, make sure it still works 
3407185f-bcbc-4c57-bc7d-cc75201615f5

## What should reviewers focus on?
* What UI cue are we expecting? This is super, super minimal
* Test coverage suggestions please :)) 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by e-wai** [comment]
Even though I definitely said otherwise at standup... I propose that this minimalistic version stands alone as its own ticket:
+ The various reset email states and moving between them can still be added once design has a more concrete idea of what that looks like (and I'm happy to create a ticket for error states and assign it to myself for when it's ready)
+ Adding Jest testing dependencies gets its own PR? It doesn't belong in this ticket's scope

Happy to debate this?

File: frontend/src/components/auth/ResetPassword.tsx [hunk 1]
```diff
@@ -2,11 +2,35 @@ import React, { useContext } from "react";
...
+  email?: string;
+}
+
+const ResetPassword = ({ email }: ResetPasswordProps) => {
   const { authenticatedUser } = useContext(AuthContext);
 
+  const isRealEmailAvailable = (): boolean => {
+    const index = email!.indexOf("@");
+
+    if (index === -1 || email!.slice(index).indexOf(".") === -1) {
```
Comment (gaoxk) [inline frontend/src/components/auth/ResetPassword.tsx:15]: nit: I suggest using a regex for emails https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript?page=1&tab=votes#tab-top

File: frontend/src/components/auth/ResetPassword.tsx [hunk 2]
```diff
@@ -2,11 +2,35 @@ import React, { useContext } from "react";
...
 
-const ResetPassword = () => {
+interface ResetPasswordProps {
+  email?: string;
+}
+
+const ResetPassword = ({ email }: ResetPasswordProps) => {
   const { authenticatedUser } = useContext(AuthContext);
 
+  const isRealEmailAvailable = (): boolean => {
```
Comment (gaoxk) [inline frontend/src/components/auth/ResetPassword.tsx:12]: nit: I suggest using an `email` parameter here and passing in the email to the function.  Makes the function's purpose crystal clear. 
i.e ` const isRealEmailAvailable = (email: string): boolean => .....`

**Review by gaoxk — CHANGES_REQUESTED** [review]

**Comment by gaoxk** [comment]
Looks good! I'm fine with moving jest to a separate PR. ui is straightforward :)

**Review by gaoxk — APPROVED** [review]
🚢 🚢 nice!

**Pull request merged**